### PR TITLE
Support for pyramids and mixed meshes for FindPointsGSLIB

### PR DESCRIPTION
**Pull request by kmittal2**
| PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#2948](https://github.com/mfem/mfem/pull/2948) | @kmittal2 | @tzanio | @brendankeith + @mlstowell + @acfisher | 4/11/22 | 5/24/22 | 5/30/22 | |

PR Checklist
    
- [x] Code builds.
- [x] Code passes `make style`.
- [x] Update `CHANGELOG`:
- [ ] All sample runs at the top of the miniapp source file work.

### CODE REVIEW

**Comment by acfisher** [comment]
Is this ready for review?

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

File: CHANGELOG [hunk 1]
```diff
@@ -19,6 +19,8 @@ Version 4.4.1 (development)
   between arbitrarily distributed and unrelated finite element meshes in a
   variationally consistent way.
 
+- Added support for mixed meshes and pyramids in GSLIBFindPoints.
```
Comment (tzanio) [inline CHANGELOG:22]: Consistency with prior styling in `CHANGELOG`

```suggestion
- Added support for mixed meshes and pyramids in GSLIB-FindPoints.
```

**Comment by acfisher** [comment]
@kmittal2 I put together a tiny zoo mesh in another branch.  Do you want to use that for testing?

**Comment by kmittal2** [comment]
Thanks for the mesh @acfisher. It is now included in `mfem/data` folder for folks who are interested in testing out new methods with a mixed mesh that has all 4 element types (hex+tet+wedge+pyramid).

**Comment by tzanio** [comment]
@brendankeith, @acfisher, @mlstowell can you please review?

File: data/tinyzoo-3d.mesh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+3
+
+elements
+4
+1 5 0 1 4 3 6 7 10 9
+1 6 4 1 5 10 7 11
+1 7 11 7 1 5 8
+1 4 2 5 1 8
+
+bounday
```
Comment (mlstowell) [inline data/tinyzoo-3d.mesh:25]: Typo in the section heading

File: data/tinyzoo-3d.mesh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+1 2 1 4 5
+1 2 1 8 7
+1 2 1 2 8
+1 2 2 5 8
+1 2 5 11 8
+1 2 7 8 11
+1 2 7 11 10
+1 3 4 10 11 5
+
+vetices
```
Comment (mlstowell) [inline data/tinyzoo-3d.mesh:42]: Another typo. Looks like someone's keyboard has a sticky 'r'.

**Comment by acfisher** [comment]
Are all of the sample runs work at this point?

File: miniapps/gslib/findpts.cpp [hunk 4]
```diff
@@ -36,6 +36,11 @@
 //    findpts -m ../../data/inline-wedge.mesh -o 3
 //    findpts -m ../../data/amr-quad.mesh -o 2
 //    findpts -m ../../data/rt-2d-q3.mesh -o 3 -mo 4 -ft 2
+//    findpts -m ../../data/square-mixed.mesh -o 2 -mo 2
```
Comment (brendankeith) [inline miniapps/gslib/findpts.cpp:39]: I think there should be an example with the `-hr` and `-pr` options. Note that `-hr` and `-pr` do not work for pyramids, so there should probably also be a check for that.

**Review by brendankeith — APPROVED** [review]

**Review by mlstowell — APPROVED** [review]

**Review by acfisher — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**